Rename `tagHistory` in `CSSSelector`

nt1m opened the pull request:
#### 0cc8d008055b8590212df7c71ded13924187a1a5
<pre>
Rename `tagHistory` in `CSSSelector`
<a href="https://bugs.webkit.org/show_bug.cgi?id=297912">https://bugs.webkit.org/show_bug.cgi?id=297912</a>
<a href="https://rdar.apple.com/159200563">rdar://159200563</a>

Reviewed by Antti Koivisto.

The tag history is actually a linked list of simple selectors within a complex selector. Make that explicit.

m_tagHistory -&gt; m_precedingInComplexSelector
tagHistory() -&gt; precedingInComplexSelector()
setTagHistory() -&gt; setPrecedingInComplexSelector()
appendTagHistory() -&gt; prependInComplexSelector()
appendTagHistoryAsRelative() -&gt; prependInComplexSelectorAsRelative()
prependTagSelector() -&gt; appendTagInComplexSelector()
releaseTagHistory() -&gt; releaseFromComplexSelector()

* Source/WebCore/css/CSSSelector.cpp:
(WebCore::selectorSpecificity):
(WebCore::CSSSelector::specificityForPage const):
(WebCore::CSSSelector::firstInCompound const):
(WebCore::CSSSelector::selectorText const):
(WebCore::CSSSelector::CSSSelector):
(WebCore::CSSSelector::visitSimpleSelectors const):
* Source/WebCore/css/CSSSelector.h:
(WebCore::CSSSelector::precedingInComplexSelector const):
(WebCore::CSSSelector::isFirstInComplexSelector const):
(WebCore::CSSSelector::isLastInComplexSelector const):
(WebCore::CSSSelector::tagHistory const): Deleted.
(WebCore::CSSSelector::isFirstInTagHistory const): Deleted.
(WebCore::CSSSelector::isLastInTagHistory const): Deleted.
* Source/WebCore/css/CSSSelectorList.cpp:
(WebCore::CSSSelectorList::CSSSelectorList):
(WebCore::CSSSelectorList::makeCopyingSimpleSelector):
(WebCore::CSSSelectorList::makeCopyingComplexSelector):
(WebCore::CSSSelectorList::listSize const):
(WebCore::forEachTagSelector):
(WebCore::CSSSelectorList::hasOnlyNestingSelector const):
* Source/WebCore/css/CSSSelectorList.h:
(WebCore::CSSSelectorList::const_iterator::operator++):
* Source/WebCore/css/SelectorChecker.cpp:
(WebCore::SelectorChecker::matchRecursively const):
(WebCore::canMatchHoverOrActiveInQuirksMode):
(WebCore::SelectorChecker::determineLinkMatchType):
* Source/WebCore/css/SelectorFilter.cpp:
(WebCore::SelectorFilter::collectSelectorHashes):
* Source/WebCore/css/StyleRule.cpp:
(WebCore::StyleRule::splitIntoMultipleRulesWithMaximumSelectorComponentCount const):
* Source/WebCore/css/parser/CSSParser.cpp:
(WebCore::CSSParser::parsePageSelector):
* Source/WebCore/css/parser/CSSSelectorParser.cpp:
(WebCore::appendImplicitSelectorPseudoClassScopeIfNeeded):
(WebCore::appendImplicitSelectorNestingParentIfNeeded):
(WebCore::CSSSelectorParser::consumeComplexSelector):
(WebCore::CSSSelectorParser::consumeRelativeScopeSelector):
(WebCore::CSSSelectorParser::consumeCompoundSelector):
(WebCore::CSSSelectorParser::prependTypeSelectorIfNeeded):
(WebCore::CSSSelectorParser::splitCompoundAtImplicitShadowCrossingCombinator):
(WebCore::CSSSelectorParser::containsUnknownWebKitPseudoElements):
* Source/WebCore/css/parser/MutableCSSSelector.cpp:
(WebCore::MutableCSSSelector::MutableCSSSelector):
(WebCore::MutableCSSSelector::~MutableCSSSelector):
(WebCore::MutableCSSSelector::leftmostSimpleSelector const):
(WebCore::MutableCSSSelector::leftmostSimpleSelector):
(WebCore::MutableCSSSelector::hasExplicitNestingParent const):
(WebCore::MutableCSSSelector::hasExplicitPseudoClassScope const):
(WebCore::selectorListMatchesPseudoElement):
(WebCore::MutableCSSSelector::insertInComplexSelector):
(WebCore::MutableCSSSelector::prependInComplexSelector):
(WebCore::MutableCSSSelector::prependInComplexSelectorAsRelative):
(WebCore::MutableCSSSelector::appendTagInComplexSelector):
(WebCore::MutableCSSSelector::releaseFromComplexSelector):
(WebCore::MutableCSSSelector::insertTagHistory): Deleted.
(WebCore::MutableCSSSelector::appendTagHistory): Deleted.
(WebCore::MutableCSSSelector::appendTagHistoryAsRelative): Deleted.
(WebCore::MutableCSSSelector::prependTagSelector): Deleted.
(WebCore::MutableCSSSelector::releaseTagHistory): Deleted.
* Source/WebCore/css/parser/MutableCSSSelector.h:
(WebCore::MutableCSSSelector::precedingInComplexSelector const):
(WebCore::MutableCSSSelector::setPrecedingInComplexSelector):
(WebCore::MutableCSSSelector::clearPrecedingInComplexSelector):
(WebCore::MutableCSSSelector::tagHistory const): Deleted.
(WebCore::MutableCSSSelector::setTagHistory): Deleted.
(WebCore::MutableCSSSelector::clearTagHistory): Deleted.
* Source/WebCore/cssjit/SelectorCompiler.cpp:
(WebCore::SelectorCompiler::constructFragmentsInternal):
* Source/WebCore/dom/SelectorQuery.cpp:
(WebCore::isSingleTagNameSelector):
(WebCore::isSingleClassNameSelector):
(WebCore::isSingleAttributeExactSelector):
(WebCore::findIdMatchingType):
(WebCore::SelectorDataList::SelectorDataList):
(WebCore::selectorForIdLookup):
(WebCore::filterRootById):
* Source/WebCore/style/HasSelectorFilter.cpp:
(WebCore::Style::HasSelectorFilter::makeKey):
* Source/WebCore/style/PageRuleCollector.cpp:
(WebCore::Style::checkPageSelectorComponents):
* Source/WebCore/style/RuleData.cpp:
(WebCore::Style::computeMatchBasedOnRuleHash):
(WebCore::Style::selectorCanMatchPseudoElement):
(WebCore::Style::determinePropertyAllowlist):
* Source/WebCore/style/RuleFeature.cpp:
(WebCore::Style::computeHasPseudoClassMatchElement):
(WebCore::Style::RuleFeatureSet::recursivelyCollectFeaturesFromSelector):
(WebCore::Style::makePseudoClassInvalidationKey):
* Source/WebCore/style/RuleSet.cpp:
(WebCore::Style::isHostSelectorMatchingInShadowTree):
(WebCore::Style::RuleSet::addRule):
* Source/WebCore/style/StyleResolver.cpp:
(WebCore::Style::viewTransitionSelector):

Canonical link: <a href="https://commits.webkit.org/299193@main">https://commits.webkit.org/299193@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/d46589a344f05e002306e436e373449115816437

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/118220 "1 style error") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/37898 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/28537 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/124379 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/70261 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/942eab48-5fd2-4b09-a5a9-a94a4fa9713d) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/120098 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/38591 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/46480 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/89709 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/59347 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/894efe11-ea12-4748-8a2d-bb311418ec11) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/121173 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/30741 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/105996 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/70205 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/36383052-e424-458e-b234-c7f14fa8db97) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/29811 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/24115 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/68045 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/100167 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/24289 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/127453 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/45123 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/34016 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/98394 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/45484 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/102217 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/98181 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/24960 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/43575 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/21563 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/41588 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/44995 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/50669 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/44455 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/47800 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/46144 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

anttijk reviewed (approved):
Looks good except for the bug title

nt1m committed:
Rename `tagHistory` in `CSSSelector`
https://bugs.webkit.org/show_bug.cgi?id=297912
rdar://159200563

Reviewed by Antti Koivisto.

The tag history is actually a linked list of simple selectors within a complex selector. Make that explicit.

m_tagHistory -> m_precedingInComplexSelector
tagHistory() -> precedingInComplexSelector()
setTagHistory() -> setPrecedingInComplexSelector()
appendTagHistory() -> prependInComplexSelector()
appendTagHistoryAsRelative() -> prependInComplexSelectorAsRelative()
prependTagSelector() -> appendTagInComplexSelector()
releaseTagHistory() -> releaseFromComplexSelector()

* Source/WebCore/css/CSSSelector.cpp:
(WebCore::selectorSpecificity):
(WebCore::CSSSelector::specificityForPage const):
(WebCore::CSSSelector::firstInCompound const):
(WebCore::CSSSelector::selectorText const):
(WebCore::CSSSelector::CSSSelector):
(WebCore::CSSSelector::visitSimpleSelectors const):
* Source/WebCore/css/CSSSelector.h:
(WebCore::CSSSelector::precedingInComplexSelector const):
(WebCore::CSSSelector::isFirstInComplexSelector const):
(WebCore::CSSSelector::isLastInComplexSelector const):
(WebCore::CSSSelector::tagHistory const): Deleted.
(WebCore::CSSSelector::isFirstInTagHistory const): Deleted.
(WebCore::CSSSelector::isLastInTagHistory const): Deleted.
* Source/WebCore/css/CSSSelectorList.cpp:
(WebCore::CSSSelectorList::CSSSelectorList):
(WebCore::CSSSelectorList::makeCopyingSimpleSelector):
(WebCore::CSSSelectorList::makeCopyingComplexSelector):
(WebCore::CSSSelectorList::listSize const):
(WebCore::forEachTagSelector):
(WebCore::CSSSelectorList::hasOnlyNestingSelector const):
* Source/WebCore/css/CSSSelectorList.h:
(WebCore::CSSSelectorList::const_iterator::operator++):
* Source/WebCore/css/SelectorChecker.cpp:
(WebCore::SelectorChecker::matchRecursively const):
(WebCore::canMatchHoverOrActiveInQuirksMode):
(WebCore::SelectorChecker::determineLinkMatchType):
* Source/WebCore/css/SelectorFilter.cpp:
(WebCore::SelectorFilter::collectSelectorHashes):
* Source/WebCore/css/StyleRule.cpp:
(WebCore::StyleRule::splitIntoMultipleRulesWithMaximumSelectorComponentCount const):
* Source/WebCore/css/parser/CSSParser.cpp:
(WebCore::CSSParser::parsePageSelector):
* Source/WebCore/css/parser/CSSSelectorParser.cpp:
(WebCore::appendImplicitSelectorPseudoClassScopeIfNeeded):
(WebCore::appendImplicitSelectorNestingParentIfNeeded):
(WebCore::CSSSelectorParser::consumeComplexSelector):
(WebCore::CSSSelectorParser::consumeRelativeScopeSelector):
(WebCore::CSSSelectorParser::consumeCompoundSelector):
(WebCore::CSSSelectorParser::prependTypeSelectorIfNeeded):
(WebCore::CSSSelectorParser::splitCompoundAtImplicitShadowCrossingCombinator):
(WebCore::CSSSelectorParser::containsUnknownWebKitPseudoElements):
* Source/WebCore/css/parser/MutableCSSSelector.cpp:
(WebCore::MutableCSSSelector::MutableCSSSelector):
(WebCore::MutableCSSSelector::~MutableCSSSelector):
(WebCore::MutableCSSSelector::leftmostSimpleSelector const):
(WebCore::MutableCSSSelector::leftmostSimpleSelector):
(WebCore::MutableCSSSelector::hasExplicitNestingParent const):
(WebCore::MutableCSSSelector::hasExplicitPseudoClassScope const):
(WebCore::selectorListMatchesPseudoElement):
(WebCore::MutableCSSSelector::insertInComplexSelector):
(WebCore::MutableCSSSelector::prependInComplexSelector):
(WebCore::MutableCSSSelector::prependInComplexSelectorAsRelative):
(WebCore::MutableCSSSelector::appendTagInComplexSelector):
(WebCore::MutableCSSSelector::releaseFromComplexSelector):
(WebCore::MutableCSSSelector::insertTagHistory): Deleted.
(WebCore::MutableCSSSelector::appendTagHistory): Deleted.
(WebCore::MutableCSSSelector::appendTagHistoryAsRelative): Deleted.
(WebCore::MutableCSSSelector::prependTagSelector): Deleted.
(WebCore::MutableCSSSelector::releaseTagHistory): Deleted.
* Source/WebCore/css/parser/MutableCSSSelector.h:
(WebCore::MutableCSSSelector::precedingInComplexSelector const):
(WebCore::MutableCSSSelector::setPrecedingInComplexSelector):
(WebCore::MutableCSSSelector::clearPrecedingInComplexSelector):
(WebCore::MutableCSSSelector::tagHistory const): Deleted.
(WebCore::MutableCSSSelector::setTagHistory): Deleted.
(WebCore::MutableCSSSelector::clearTagHistory): Deleted.
* Source/WebCore/cssjit/SelectorCompiler.cpp:
(WebCore::SelectorCompiler::constructFragmentsInternal):
* Source/WebCore/dom/SelectorQuery.cpp:
(WebCore::isSingleTagNameSelector):
(WebCore::isSingleClassNameSelector):
(WebCore::isSingleAttributeExactSelector):
(WebCore::findIdMatchingType):
(WebCore::SelectorDataList::SelectorDataList):
(WebCore::selectorForIdLookup):
(WebCore::filterRootById):
* Source/WebCore/style/HasSelectorFilter.cpp:
(WebCore::Style::HasSelectorFilter::makeKey):
* Source/WebCore/style/PageRuleCollector.cpp:
(WebCore::Style::checkPageSelectorComponents):
* Source/WebCore/style/RuleData.cpp:
(WebCore::Style::computeMatchBasedOnRuleHash):
(WebCore::Style::selectorCanMatchPseudoElement):
(WebCore::Style::determinePropertyAllowlist):
* Source/WebCore/style/RuleFeature.cpp:
(WebCore::Style::computeHasPseudoClassMatchElement):
(WebCore::Style::RuleFeatureSet::recursivelyCollectFeaturesFromSelector):
(WebCore::Style::makePseudoClassInvalidationKey):
* Source/WebCore/style/RuleSet.cpp:
(WebCore::Style::isHostSelectorMatchingInShadowTree):
(WebCore::Style::RuleSet::addRule):
* Source/WebCore/style/StyleResolver.cpp:
(WebCore::Style::viewTransitionSelector):

Canonical link: https://commits.webkit.org/299193@main
